litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/index.tsx

```diff
@@ -0,0 +1,263 @@
+import { useQuery } from '@apollo/client';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@material-ui/core';
+import moment from 'moment';
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { WORKFLOW_DETAILS, WORKFLOW_EVENTS } from '../../../graphql';
+import {
+  Workflow,
+  WorkflowDataVars,
+  WorkflowSubscription,
+} from '../../../models/graphql/workflowData';
+import { RootState } from '../../../redux/reducers';
+import HeaderSection from './HeaderSection';
+import useStyles from './styles';
+
+interface FilterOptions {
+  search: string;
+  status: string;
+  cluster: string;
+}
+
+interface PaginationData {
+  pageNo: number;
+  rowsPerPage: number;
+}
+
+interface SortData {
+  lastRun: { sort: boolean; ascending: boolean };
+  name: { sort: boolean; ascending: boolean };
+  noOfSteps: { sort: boolean; ascending: boolean };
+}
+
+interface DateData {
+  dateValue: string;
+  fromDate: string;
+  toDate: string;
+}
+
+const BrowseWorkflow = () => {
+  const classes = useStyles();
+  const selectedProjectID = useSelector(
+    (state: RootState) => state.userData.selectedProjectID
+  );
+
+  // Query to get workflows
+  const { subscribeToMore, data } = useQuery<Workflow, WorkflowDataVars>(
+    WORKFLOW_DETAILS,
+    {
+      variables: { projectID: selectedProjectID },
+      fetchPolicy: 'cache-and-network',
+    }
+  );
+
+  // Using subscription to get realtime data
+  useEffect(() => {
+    subscribeToMore<WorkflowSubscription>({
+      document: WORKFLOW_EVENTS,
+      variables: { projectID: selectedProjectID },
+      updateQuery: (prev, { subscriptionData }) => {
+        if (!subscriptionData.data) return prev;
+        const modifiedWorkflows = prev.getWorkFlowRuns.slice();
```

Comment:
why is `getWorkFlowRuns` used for connecting target clusters? @oumkale cc: @gdsoumya @rajdas98